Polyhedron/demo/Polyhedron/Plugins/Alpha_wrap_3/Alpha_wrap_3_plugin.cpp

```diff
@@ -275,35 +386,138 @@ class Polyhedron_demo_alpha_wrap_3_plugin
   }
 
   void on_offsetValue_changed(double)
   {
     QSignalBlocker block(ui.relativeOffsetValue);
-    ui.relativeOffsetValue->setValue(wrap_bbox_diag_length / ui.offsetValue->value());
+    ui.relativeOffsetValue->setValue(m_wrap_bbox_diag_length / ui.offsetValue->value());
   }
 
   void on_relativeOffsetValue_changed(double)
   {
     QSignalBlocker block(ui.offsetValue);
-    ui.offsetValue->setValue(wrap_bbox_diag_length / ui.relativeOffsetValue->value());
+    ui.offsetValue->setValue(m_wrap_bbox_diag_length / ui.relativeOffsetValue->value());
   }
 
   void update_iteration_snapshot_checkbox()
   {
     ui.snapshotIterations->setCheckable(ui.visualizeIterations->isChecked());
   }
 
+  // This is for the visualization
+  void update_iterative_wrap_item(int sid)
+  {
+    if(m_iterative_wrap_item == nullptr)
+      return;
+
+    // Update the wrap item's visualization
+    m_iterative_wrap_item->load(*m_iter_points, *m_iter_faces, *m_iter_fcolors, *m_iter_vcolors);
+    m_iterative_wrap_item->setName(QString("Iterative wrap #%1").arg(sid));
+    m_iterative_wrap_item->setAlpha(255 / 2);
+    m_iterative_wrap_item->setRenderingMode(FlatPlusEdges);
+
+    m_iterative_wrap_item->invalidateOpenGLBuffers();
+    m_iterative_wrap_item->redraw();
+    m_iterative_wrap_item->itemChanged();
+
+    // Refresh the view
+    CGAL::Three::Viewer_interface* viewer = CGAL::Three::Three::activeViewer();
+    viewer->update();
+
+    CGAL::Three::Three::getWaitCondition()->wakeAll();
+
+    if(m_do_snapshot)
+    {
+      std::stringstream oss;
+      oss << "Wrap_iteration-" << sid << ".png" << std::ends;
+      QString filename = QString::fromStdString(oss.str().c_str());
+
+      viewer->saveSnapshot(filename, 1920, 1080, true /*expand*/, 2.0 /*oversampling*/);
+    }
+  }
+
+  void finish_iterative_wrap_item(int sid)
+  {
+    if(m_iterative_wrap_item == nullptr)
+      return;
+
+    if(m_do_snapshot)
+    {
+      m_iterative_wrap_item->setName(QString("Iterative wrap #%1").arg(sid));
+
+      m_iterative_wrap_item->setAlpha(255);
+      m_iterative_wrap_item->invalidateOpenGLBuffers();
+      m_iterative_wrap_item->redraw();
+      m_iterative_wrap_item->itemChanged();
+
+      // Refresh the view
+      CGAL::Three::Viewer_interface* viewer = CGAL::Three::Three::activeViewer();
+      viewer->update();
+
+      std::stringstream oss;
+      oss << "Wrap_iteration-" << sid << ".png" << std::ends;
+      QString filename = QString::fromStdString(oss.str().c_str());
+
+      viewer->saveSnapshot(filename);
+    }
+
+    CGAL_assertion(m_iterative_wrap_item);
+    scene->erase(scene->item_id(m_iterative_wrap_item));
+  }
+
+  void reset_iterative_wrap_item()
+  {
+    m_iterative_wrap_item = nullptr;
+  }
+
+  // This is for the message box and thread interruption
+  void wrapping_done(Wrapper_thread* wrapper_thread)
+  {
+    Scene_surface_mesh_item* wrap_item = new Scene_surface_mesh_item(wrapper_thread->wrap);
```

Comment:
To avoid a copy: move the `wrap` surface mesh, that is useless after that call:
```suggestion
    Scene_surface_mesh_item* wrap_item = new Scene_surface_mesh_item(std::move(wrapper_thread->wrap));
```